cfg/config.go

```diff
@@ -319,20 +321,55 @@ func (c Config) validateConfig() error {
 		return errors.New("GitHub token required")
 	}
 
-	jUser := c.cmdConfig.GetString("jira-user")
-	if jUser == "" {
-		return errors.New("Jira username required")
-	}
+	if c.cmdConfig.GetString("jira-user") != "" || c.cmdConfig.GetString("jira-pass") != "" {
+		c.log.Debug("Using Basic Authentication")
 
-	jPass := c.cmdConfig.GetString("jira-pass")
-	if jPass == "" {
-		fmt.Print("Enter your JIRA password: ")
-		bytePass, err := terminal.ReadPassword(int(syscall.Stdin))
-		if err != nil {
-			return errors.New("Jira password required")
+		jUser := c.cmdConfig.GetString("jira-user")
+		if jUser == "" {
+			return errors.New("JIRA username required")
+		}
+
+		jPass := c.cmdConfig.GetString("jira-pass")
+		if jPass == "" {
+			fmt.Print("Enter your JIRA password: ")
+			bytePass, err := terminal.ReadPassword(int(syscall.Stdin))
+			if err != nil {
+				return errors.New("JIRA password required")
+			}
+			fmt.Println()
+			c.cmdConfig.Set("jira-pass", string(bytePass))
+		}
+	} else {
+		if c.cmdConfig.GetString("jira-token") != "" || c.cmdConfig.GetString("jira-secret") != "" {
+			c.log.Debug("Using configured OAuth access token")
+
+			token := c.cmdConfig.GetString("jira-token")
+			if token == "" {
+				return errors.New("JIRA access token required")
+			}
+
+			secret := c.cmdConfig.GetString("jira-secret")
+			if secret == "" {
+				return errors.New("JIRA access token secret required")
+			}
+
+			c.log.Debug("Using OAuth; will perform handshake")
```

Comment:
The handshake is not performed here, so I think this debug statement might be a little confusing when you're actually digging through oauth problems.